providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
I wouldn't execute this definition during compilation. It should be lazily evaluated at run time.
Well, I looked through some of the selinux sources and I think its save to assume they use basic POSIX regexes.
We could narrow the find down by using the get_restorecon_path I suggested above. That should definitely speed things up a lot. That should also remove most "access denied" problems.

Giving up is not really an option in my eyes. We do have solutions, we just don't like them. ;)